modules/Matting/Global-Sampling-Method-for-Alpha-Matting/guidedfilter.cpp

```diff
@@ -0,0 +1,212 @@
+#include "guidedfilter.h"
+
+static cv::Mat boxfilter(const cv::Mat &I, int r)
```

Comment:
Same for boxfilter, it is already in OpenCV 

https://docs.opencv.org/master/d4/d86/group__imgproc__filter.html#gad533230ebf2d42509547d514f7d3fbc3